packages/grid/x-data-grid-premium/src/hooks/features/rowGrouping/gridRowGroupingUtils.ts

```diff
@@ -65,56 +71,69 @@ const shouldApplyFilterItemOnGroup = (columnField: string, node: GridRowTreeNode
 export const filterRowTreeFromGroupingColumns = (
   params: FilterRowTreeFromTreeDataParams,
 ): Omit<GridFilterState, 'filterModel'> => {
-  const { rowTree, isRowMatchingFilters } = params;
+  const { rowTree, isRowMatchingFilters, filterModel } = params;
   const visibleRowsLookup: Record<GridRowId, boolean> = {};
   const filteredRowsLookup: Record<GridRowId, boolean> = {};
   const filteredDescendantCountLookup: Record<GridRowId, number> = {};
 
   const filterTreeNode = (
     node: GridRowTreeNodeConfig,
-    areAncestorsPassingChildren: boolean,
     areAncestorsExpanded: boolean,
+    ancestorsResults: any[],
   ): number => {
-    let isMatchingFilters: boolean;
-    if (!isRowMatchingFilters) {
-      isMatchingFilters = true;
-    } else {
+    let isPassingFiltering: boolean | undefined;
+    let filterResults: {
+      passingFilterItems: null | GridFilterResult;
+      passingQuickFilterValues: null | GridFilterResult;
+    } = {
+      passingFilterItems: null,
+      passingQuickFilterValues: null,
+    };
+
+    if (isRowMatchingFilters && node.position !== 'footer') {
       const shouldApplyItem = node.isAutoGenerated
         ? (columnField: string) => shouldApplyFilterItemOnGroup(columnField, node)
         : undefined;
 
-      isMatchingFilters = isRowMatchingFilters(node.id, shouldApplyItem);
+      filterResults = isRowMatchingFilters(node.id, shouldApplyItem);
+    } else {
+      isPassingFiltering = true;
     }
 
     let filteredDescendantCount = 0;
     node.children?.forEach((childId) => {
       const childNode = rowTree[childId];
       const childSubTreeSize = filterTreeNode(
         childNode,
-        areAncestorsPassingChildren && isMatchingFilters,
         areAncestorsExpanded && !!node.childrenExpanded,
+        [...ancestorsResults, filterResults],
       );
       filteredDescendantCount += childSubTreeSize;
     });
 
-    let shouldPassFilters: boolean;
-    if (!areAncestorsPassingChildren) {
-      shouldPassFilters = false;
-    } else if (node.children?.length) {
-      shouldPassFilters = isMatchingFilters && filteredDescendantCount > 0;
-    } else {
-      shouldPassFilters = isMatchingFilters;
+    if (isPassingFiltering === undefined) {
+      if (node.children?.length) {
+        // If it has a tree structure, parent pass filter if they have at least one children passing
+        isPassingFiltering = filteredDescendantCount > 0;
+      } else {
+        const allResults = [...ancestorsResults, filterResults];
+        isPassingFiltering = passFilterLogic(
+          allResults.map((result) => result.passingFilterItems),
+          allResults.map((result) => result.passingQuickFilterValues),
+          filterModel,
```

Comment:
I don't know. It seems more logical that the strategy chose which row result is passed to the `passFilterLogic`, and the `passFilterLogic` applies them without smart features.

For example a strategy could chose to apply filter item on all row, and quick filter only on leaf

Reply:
It will still be possible to do:
```tsx
allResults,map(result => {
  const newResult = { ...result };
  if (something) {
    newResult.passQuickFilterLogic = null
  }
})
```
before passing it to `passFilterLogic`.

But feel free to keep it the way it is - performance gains won't be huge here anyway, since `allResults` would usually have only few elements at most.